crcsim/experiment/prepare.py

```diff
@@ -123,6 +123,27 @@ def transform(params):
 
     return transform
 
+def transform_delayed_onset_compliance(age_ranges, compliance_rates, test) -> Callable:
+    def transform(params):
+        params["delayed_onset_compliance"] = {
+            "age_ranges": age_ranges,
+            "compliance_rates": compliance_rates,
+            "test": test
+        }
+    return transform
+
+def transform_random_compliance(test: str) -> Callable:
+    def transform(params):
+        params["random_compliance"] = {
+            "test": test
+        }
+        initial_compliance_rate = random.uniform(0.5, 0.9)
+        params["tests"][test]["initial_compliance_rate"] = initial_compliance_rate
+        for year in range(1, 10):
+            conditional_compliance_rate = random.uniform(0.5, 0.9)
+            params["tests"][test][f"compliance_rate_year_{year}"] = conditional_compliance_rate
+
+    return transform
```

Comment:
I'm not sure what the goal is with these transform functions. It looks like you're creating some new parameters `delayed_onset_compliance` and `random_compliance`, but those parameters aren't going to do anything unless you make changes to the model to use them.

Rather than doing that, I think a better approach would be to make transform functions that adjust the existing `compliance_rate_given_prev_compliant` and `compliance_rate_given_not_prev_compliant` parameters for each test.